PICMI_Python/particles.py

```diff
@@ -27,8 +28,19 @@ class PICMI_Species(_ClassWithInit):
       - particle_shape: Particle shape used for deposition and gather ; if None, the default from the `Simulation` object will be used. Possible values are 'NGP', 'linear', 'quadratic', 'cubic'
     """
 
+    methods_list = ['Boris' , 'Vay', 'Higuera-Cary', 'Li', 'free-streaming', 'LLRK4']
+
     def __init__(self, particle_type=None, name=None, charge_state=None, charge=None, mass=None,
                  initial_distribution=None, particle_shape=None, density_scale=None, **kw):
+
+
+        assert method is None or method in PICMI_Species.methods_list or method.startswith('other:'), \
+            Exception('method must starts with either other:, or be one of the following '+', '.join(PICMI_Species.methods_list))    
+
+        if (method is None):
```

Comment:
The standard so far in PICMI is to leave input parameters as None if they are not specified (and not set any specific defaults). This should only be `self.method = method`.